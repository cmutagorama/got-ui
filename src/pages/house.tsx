import React, { FC } from 'react'
import RootPage from './root'

const House: FC = (props) => {
	return (
		<RootPage header='House'>
			<h3>House</h3>
		</RootPage>
	)
}

export default House;