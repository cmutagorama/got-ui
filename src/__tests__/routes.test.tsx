import * as React from "react";
import { create } from "react-test-renderer";
import {
	MemoryRouter,
	Routes,
	Route,
} from "react-router-dom";
import Home from "../pages/home";
import House from "../pages/house";
import { NoMatch } from "../pages/nomatch";
import Books from "../pages/books";
import Characters from "../pages/characters";

describe("Routes", () => {
	let consoleWarn: jest.SpyInstance;
  let consoleError: jest.SpyInstance;
  beforeEach(() => {
    consoleWarn = jest.spyOn(console, "warn").mockImplementation(() => {});
    consoleError = jest.spyOn(console, "error").mockImplementation(() => {});
  });

  afterEach(() => {
    consoleWarn.mockRestore();
    consoleError.mockRestore();
	});

  it("should render home page", () => {
    let renderer = create(
      <MemoryRouter initialEntries={["/"]}>
        <Routes>
          <Route path="/" element={<Home />}>
          </Route>
        </Routes>
      </MemoryRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
	});
	
	it("should render house details page", () => {
    let renderer = create(
      <MemoryRouter initialEntries={["/house/1"]}>
        <Routes>
          <Route path="/house/:id" element={<House />}>
          </Route>
        </Routes>
      </MemoryRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
	});

	it("should render characters page", () => {
    let renderer = create(
      <MemoryRouter initialEntries={["/characters"]}>
        <Routes>
          <Route path="/characters" element={<Characters />}>
          </Route>
        </Routes>
      </MemoryRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
	});
	
	it("should render books page", () => {
    let renderer = create(
      <MemoryRouter initialEntries={["/books"]}>
        <Routes>
          <Route path="/books" element={<Books />}>
          </Route>
        </Routes>
      </MemoryRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
	});
	
	it("should render 404 page", () => {
    let renderer = create(
      <MemoryRouter initialEntries={["/wrong-page"]}>
        <Routes>
          <Route path="*" element={<NoMatch />}>
          </Route>
        </Routes>
      </MemoryRouter>
    );

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});