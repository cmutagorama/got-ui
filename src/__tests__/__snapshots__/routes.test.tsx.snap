// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render 404 page 1`] = `
<div
  className="flex flex-col"
>
  <nav
    className="relative flex flex-wrap items-center justify-between px-2 py-3 mb-3 bg-gray-900"
  >
    <div
      className="container flex flex-wrap items-center justify-between px-4 mx-auto"
    >
      <div
        className="relative flex justify-between w-full lg:w-auto lg:static lg:block lg:justify-start"
      >
        <a
          className="inline-block py-2 mr-4 text-sm font-bold leading-relaxed text-white uppercase whitespace-nowrap"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="GOT logo"
            src="logo.svg"
            style={
              Object {
                "height": "40px",
                "width": "300px",
              }
            }
          />
        </a>
        <button
          className="block px-3 py-1 text-xl leading-none text-white bg-transparent border border-transparent border-solid rounded outline-none cursor-pointer lg:hidden focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        className="lg:flex flex-grow items-center hidden"
        id="example-navbar-danger"
      >
        <ul
          className="flex flex-col list-none lg:flex-row lg:ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/houses"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Houses
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/characters"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Characters
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/books"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Books
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <section
    className="flex items-center h-full p-16 dark:bg-gray-900 dark:text-gray-100"
  >
    <div
      className="container flex flex-col items-center justify-center px-5 mx-auto my-8"
    >
      <div
        className="max-w-md text-center"
      >
        <h2
          className="mb-8 font-extrabold text-9xl dark:text-gray-600"
        >
          <span
            className="sr-only"
          >
            Error
          </span>
          404
        </h2>
        <p
          className="text-2xl font-semibold md:text-3xl"
        >
          Sorry, we couldn't find this page.
        </p>
        <p
          className="mt-4 mb-8 dark:text-gray-400"
        >
          But dont worry, you can find plenty of other things on our homepage.
        </p>
        <a
          className="px-8 py-3 font-semibold rounded dark:bg-violet-400 dark:text-gray-900"
          href="/"
          onClick={[Function]}
        >
          Back to homepage
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`Routes should render books page 1`] = `
<div
  className="flex flex-col"
>
  <nav
    className="relative flex flex-wrap items-center justify-between px-2 py-3 mb-3 bg-gray-900"
  >
    <div
      className="container flex flex-wrap items-center justify-between px-4 mx-auto"
    >
      <div
        className="relative flex justify-between w-full lg:w-auto lg:static lg:block lg:justify-start"
      >
        <a
          className="inline-block py-2 mr-4 text-sm font-bold leading-relaxed text-white uppercase whitespace-nowrap"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="GOT logo"
            src="logo.svg"
            style={
              Object {
                "height": "40px",
                "width": "300px",
              }
            }
          />
        </a>
        <button
          className="block px-3 py-1 text-xl leading-none text-white bg-transparent border border-transparent border-solid rounded outline-none cursor-pointer lg:hidden focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        className="lg:flex flex-grow items-center hidden"
        id="example-navbar-danger"
      >
        <ul
          className="flex flex-col list-none lg:flex-row lg:ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/houses"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Houses
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/characters"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Characters
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/books"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Books
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="flex flex-col px-8 mt-6"
  >
    <div
      className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
    >
      <div
        className="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8"
      >
        <div
          className="overflow-hidden border-b border-gray-200 rounded-md shadow-md"
        >
          <table
            className="min-w-full overflow-x-scroll divide-y divide-gray-200"
          >
            <thead
              className="bg-gray-50"
            >
              <tr>
                <th
                  className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
                  scope="col"
                >
                  Name
                </th>
                <th
                  className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
                  scope="col"
                >
                  Publisher
                </th>
                <th
                  className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
                  scope="col"
                >
                  ISBN
                </th>
                <th
                  className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
                  scope="col"
                >
                  Country
                </th>
                <th
                  className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
                  scope="col"
                >
                  Media Type
                </th>
                <th
                  className="px-6 py-3 text-xs font-medium tracking-wider text-left text-gray-500 uppercase"
                  scope="col"
                >
                  Released Date
                </th>
              </tr>
            </thead>
            <tbody
              className="bg-white divide-y divide-gray-200"
            />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Routes should render characters page 1`] = `
<div
  className="flex flex-col"
>
  <nav
    className="relative flex flex-wrap items-center justify-between px-2 py-3 mb-3 bg-gray-900"
  >
    <div
      className="container flex flex-wrap items-center justify-between px-4 mx-auto"
    >
      <div
        className="relative flex justify-between w-full lg:w-auto lg:static lg:block lg:justify-start"
      >
        <a
          className="inline-block py-2 mr-4 text-sm font-bold leading-relaxed text-white uppercase whitespace-nowrap"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="GOT logo"
            src="logo.svg"
            style={
              Object {
                "height": "40px",
                "width": "300px",
              }
            }
          />
        </a>
        <button
          className="block px-3 py-1 text-xl leading-none text-white bg-transparent border border-transparent border-solid rounded outline-none cursor-pointer lg:hidden focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        className="lg:flex flex-grow items-center hidden"
        id="example-navbar-danger"
      >
        <ul
          className="flex flex-col list-none lg:flex-row lg:ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/houses"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Houses
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/characters"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Characters
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/books"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Books
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="grid gap-4 px-4 py-3 md:grid-cols-4"
  />
</div>
`;

exports[`Routes should render home page 1`] = `
<div
  className="flex flex-col"
>
  <nav
    className="relative flex flex-wrap items-center justify-between px-2 py-3 mb-3 bg-gray-900"
  >
    <div
      className="container flex flex-wrap items-center justify-between px-4 mx-auto"
    >
      <div
        className="relative flex justify-between w-full lg:w-auto lg:static lg:block lg:justify-start"
      >
        <a
          className="inline-block py-2 mr-4 text-sm font-bold leading-relaxed text-white uppercase whitespace-nowrap"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="GOT logo"
            src="logo.svg"
            style={
              Object {
                "height": "40px",
                "width": "300px",
              }
            }
          />
        </a>
        <button
          className="block px-3 py-1 text-xl leading-none text-white bg-transparent border border-transparent border-solid rounded outline-none cursor-pointer lg:hidden focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        className="lg:flex flex-grow items-center hidden"
        id="example-navbar-danger"
      >
        <ul
          className="flex flex-col list-none lg:flex-row lg:ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/houses"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Houses
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/characters"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Characters
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/books"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Books
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="p-4"
  >
    <h3>
      Landing page
    </h3>
  </div>
</div>
`;

exports[`Routes should render house details page 1`] = `
<div
  className="flex flex-col"
>
  <nav
    className="relative flex flex-wrap items-center justify-between px-2 py-3 mb-3 bg-gray-900"
  >
    <div
      className="container flex flex-wrap items-center justify-between px-4 mx-auto"
    >
      <div
        className="relative flex justify-between w-full lg:w-auto lg:static lg:block lg:justify-start"
      >
        <a
          className="inline-block py-2 mr-4 text-sm font-bold leading-relaxed text-white uppercase whitespace-nowrap"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="GOT logo"
            src="logo.svg"
            style={
              Object {
                "height": "40px",
                "width": "300px",
              }
            }
          />
        </a>
        <button
          className="block px-3 py-1 text-xl leading-none text-white bg-transparent border border-transparent border-solid rounded outline-none cursor-pointer lg:hidden focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        className="lg:flex flex-grow items-center hidden"
        id="example-navbar-danger"
      >
        <ul
          className="flex flex-col list-none lg:flex-row lg:ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/houses"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Houses
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/characters"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Characters
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/books"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Books
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="px-6 overflow-hidden bg-white shadow sm:rounded-lg"
  >
    <div
      className="px-2 py-3 md:px-4"
    >
      <button
        className="inline-flex items-center px-3 py-1 mt-4 text-base bg-gray-200 border-0 rounded focus:outline-none hover:bg-gray-300 md:mt-0"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="w-4 h-4 mr-2"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M20.25 12a.75.75 0 01-.75.75H6.31l5.47 5.47a.75.75 0 11-1.06 1.06l-6.75-6.75a.75.75 0 010-1.06l6.75-6.75a.75.75 0 111.06 1.06l-5.47 5.47H19.5a.75.75 0 01.75.75z"
            fillRule="evenodd"
          />
        </svg>
        Go back
      </button>
    </div>
    <div
      className="px-2 py-5 md:px-4 sm:px-6"
    >
      <h3
        className="text-lg font-medium leading-6 text-gray-900"
      >
        House Details
      </h3>
      <p
        className="max-w-2xl mt-1 text-sm text-gray-500"
      >
        Read all information
      </p>
    </div>
    <div
      className="px-2 border-t border-gray-200 md:px-4"
    >
      <dl />
    </div>
  </div>
</div>
`;

exports[`Routes should render houses page 1`] = `
<div
  className="flex flex-col"
>
  <nav
    className="relative flex flex-wrap items-center justify-between px-2 py-3 mb-3 bg-gray-900"
  >
    <div
      className="container flex flex-wrap items-center justify-between px-4 mx-auto"
    >
      <div
        className="relative flex justify-between w-full lg:w-auto lg:static lg:block lg:justify-start"
      >
        <a
          className="inline-block py-2 mr-4 text-sm font-bold leading-relaxed text-white uppercase whitespace-nowrap"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="GOT logo"
            src="logo.svg"
            style={
              Object {
                "height": "40px",
                "width": "300px",
              }
            }
          />
        </a>
        <button
          className="block px-3 py-1 text-xl leading-none text-white bg-transparent border border-transparent border-solid rounded outline-none cursor-pointer lg:hidden focus:outline-none"
          onClick={[Function]}
          type="button"
        >
          <svg
            aria-hidden="true"
            className="w-6 h-6"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
              fillRule="evenodd"
            />
          </svg>
        </button>
      </div>
      <div
        className="lg:flex flex-grow items-center hidden"
        id="example-navbar-danger"
      >
        <ul
          className="flex flex-col list-none lg:flex-row lg:ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/houses"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Houses
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/characters"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Characters
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="flex items-center px-3 py-2 text-xs font-bold leading-snug text-white uppercase hover:opacity-75"
              href="/books"
              onClick={[Function]}
            >
              <span
                className="ml-2"
              >
                Books
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="grid gap-4 px-4 py-3 md:grid-cols-4"
  >
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full max-w-sm p-4 mx-auto border border-blue-300 rounded-md shadow"
    >
      <div
        className="flex space-x-4 animate-pulse"
      >
        <div
          className="w-10 h-10 rounded-full bg-slate-700"
        />
        <div
          className="flex-1 py-1 space-y-6"
        >
          <div
            className="h-2 rounded bg-slate-700"
          />
          <div
            className="space-y-3"
          >
            <div
              className="grid grid-cols-3 gap-4"
            >
              <div
                className="h-2 col-span-2 rounded bg-slate-700"
              />
              <div
                className="h-2 col-span-1 rounded bg-slate-700"
              />
            </div>
            <div
              className="h-2 rounded bg-slate-700"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex flex-col items-center px-4 py-3 space-x-0 md:flex-row md:space-x-4"
  >
    <div
      className="flex flex-row items-center justify-center space-x-3"
    >
      <p
        className="text-xs"
      >
        Items per page
      </p>
      <div
        className="flex flex-row"
      >
        <div
          className="relative mt-1"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="relative w-full py-2 pl-3 pr-10 text-left bg-white rounded-lg shadow-md cursor-default focus:outline-none focus-visible:border-indigo-500 focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75 focus-visible:ring-offset-2 focus-visible:ring-offset-orange-300 sm:text-sm"
            data-headlessui-state=""
            disabled={false}
            id="headlessui-listbox-button-:r0:"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            type="button"
          >
            <span
              className="block truncate"
            >
              10
            </span>
            <span
              className="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
            >
              <svg
                aria-hidden="true"
                className="w-5 h-5 text-gray-400"
                fill="currentColor"
                viewBox="0 0 20 20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clipRule="evenodd"
                  d="M10 3a.75.75 0 01.55.24l3.25 3.5a.75.75 0 11-1.1 1.02L10 4.852 7.3 7.76a.75.75 0 01-1.1-1.02l3.25-3.5A.75.75 0 0110 3zm-3.76 9.2a.75.75 0 011.06.04l2.7 2.908 2.7-2.908a.75.75 0 111.1 1.02l-3.25 3.5a.75.75 0 01-1.1 0l-3.25-3.5a.75.75 0 01.04-1.06z"
                  fillRule="evenodd"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="flex items-center justify-between w-full h-10 text-sm select-none md:justify-center"
    >
      <button
        className="flex items-center mr-2 text-gray-500 opacity-50 hover:text-gray-600 focus:outline-none"
        disabled={true}
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="w-4 h-4 mr-2"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M11.03 3.97a.75.75 0 010 1.06l-6.22 6.22H21a.75.75 0 010 1.5H4.81l6.22 6.22a.75.75 0 11-1.06 1.06l-7.5-7.5a.75.75 0 010-1.06l7.5-7.5a.75.75 0 011.06 0z"
            fillRule="evenodd"
          />
        </svg>
        Previous
      </button>
      <div
        className="items-center justify-center hidden mx-5 md:flex"
      >
        <span
          className="flex items-center justify-center w-10 h-10 rounded-full cursor-pointer bg-blue-50 text-blue-600"
          onClick={[Function]}
        >
          1
        </span>
        <span
          className="flex items-center justify-center w-10 h-10 rounded-full cursor-pointer text-gray-500"
          onClick={[Function]}
        >
          2
        </span>
        <span
          className="flex items-center justify-center w-10 h-10 rounded-full cursor-pointer text-gray-500"
          onClick={[Function]}
        >
          3
        </span>
        <span
          className="w-10 px-0.5 text-center"
        >
          ...
        </span>
        <span
          className="flex items-center justify-center w-10 h-10 rounded-full cursor-pointer text-gray-500"
          onClick={[Function]}
        >
          9
        </span>
        <span
          className="flex items-center justify-center w-10 h-10 rounded-full cursor-pointer text-gray-500"
          onClick={[Function]}
        >
          10
        </span>
      </div>
      <button
        className="flex items-center mr-2 text-gray-500 cursor-pointer hover:text-gray-600 focus:outline-none"
        disabled={false}
        onClick={[Function]}
      >
        Next
        <svg
          aria-hidden="true"
          className="w-4 h-4 ml-2"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M12.97 3.97a.75.75 0 011.06 0l7.5 7.5a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 11-1.06-1.06l6.22-6.22H3a.75.75 0 010-1.5h16.19l-6.22-6.22a.75.75 0 010-1.06z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
